clang/lib/Sema/TreeTransform.h

```diff
@@ -4006,6 +4007,37 @@ class TreeTransform {
                                       NumExpansions);
   }
 
+  void RebuildLambdaExprImpl(SourceLocation StartLoc, SourceLocation EndLoc,
+                             LambdaScopeInfo *LSI) {}
+
+  ExprResult RebuildLambdaExpr(SourceLocation StartLoc, SourceLocation EndLoc,
+                               LambdaScopeInfo *LSI) {
+    CXXRecordDecl *Class = LSI->Lambda;
+    CXXMethodDecl *CallOperator = LSI->CallOperator;
+    CallOperator->setLexicalDeclContext(Class);
+    Decl *TemplateOrNonTemplateCallOperatorDecl =
+        CallOperator->getDescribedFunctionTemplate()
+            ? CallOperator->getDescribedFunctionTemplate()
+            : cast<Decl>(CallOperator);
+    // FIXME: Is this really the best choice? Keeping the lexical decl context
+    // set as CurContext seems more faithful to the source.
+    TemplateOrNonTemplateCallOperatorDecl->setLexicalDeclContext(Class);
+
+    getDerived().RebuildLambdaExprImpl(StartLoc, EndLoc, LSI);
```

Comment:
We don't seem to do anything interesting there

Reply:
@cor3ntin We call into `TemplateInstantiator::RebuildLambdaExprImpl(),` which substitutes for default arguments. I intend to set the flag after the substitution, as it's likely we could lose the unexpanded pack afterward.

Re "Why not just calling BuildLambdaExpr first?":
`BuildLambdaExpr()` would consume the `ContainsUnexpandedParameterPack` flag and use it as part of the dependency of the `LambdaExpr`. And what we used to do was to substitute for default arguments after `BuildLambdaExpr()`, which makes us miss the chance of altering the flag.

As a result, we would have such a call chain:
```
RebuildLambdaExpr() -> BuildLambdaExprImpl() (which actually performs the default argument substitution) -> BuildLambdaExpr()
```

Reply:
@cor3ntin Aha, I've figured out a way to simplify this `RebuildLambdaExpr()` thing: I think it's unnecessary to change the new method decl's lexical context to CurContext (we do so in parsing because we want to align the Scope nesting with the lexical nesting; however this is just what needs to be respected in parsing) and this way, we could just get rid of these setter calls.